docs/source/developers/python.rst

```diff
@@ -248,8 +243,8 @@ folder as the repositories and a target installation folder:
 
 .. code-block:: shell
 
-   virtualenv pyarrow
-   source ./pyarrow/bin/activate
+   virtualenv -p python3.9 pyarrow-dev
```

Comment:
I didn't know this flag existed.
For the record, it is sufficient these days to use the built-in [venv](https://docs.python.org/3/library/venv.html) module, while `virtualenv` needs to be installed separately. That said, I suppose many developers are more used to `virtualenv`.

Reply:
(disclaimer: I am not very familiar with those tools, never use them myself) 
I think virtualenv additionally allows to create an environment with a different python version? (in which case it might still be useful to use it here in the example)

Reply:
You would write `python3.9 -m venv pyarrow-dev`

Reply:
But that requires you have python 3.9 installed? (and eg if you install python from python.org, I think you already are linked to python 3.10 by default)

Reply:
You mean virtualenv will install the desired Python version for you?

Reply:
> You mean virtualenv will install the desired Python version for you?

That was my assumption when writing that, but it might be you can only specify python versions you already have installed? (again, I am not familiar with virtualenv, so not the best person to know what we should recommend here :))

Reply:
https://virtualenv.pypa.io/en/latest/cli_interface.html#p